pcdsutils/log.py

```diff
@@ -321,3 +338,256 @@ def centralized_logging_enabled() -> bool:
     """Returns True if centralized logging should be enabled."""
     fqdn = get_fully_qualified_domain_name()
     return any(fqdn.endswith(domain) for domain in ALLOWED_LOG_DOMAINS)
+
+
+warnings_logger = logging.getLogger(f'{__name__}.warnings')
+
+
+def log_warning_handler(
+    message: Warning,
+    category: type[Warning],
+    filename: str,
+    lineno: int,
+    file: typing.Optional[typing.TextIO] = None,
+    line: typing.Optional[str] = None,
+    logger: logging.Logger = warnings_logger,
+):
+    """
+    Warning handler that redirects all of the warnings to a logger.
+
+    This can be used as a drop-in replacement for warnings.showwarning to
+    redirect unfiltered warnings into the logging stream.
+
+    Rather than duplicate the warning display text, I've opted to simplify it
```

Comment:
```suggestion
    Rather than duplicate the warning display text, this handler opts to simplify it
```
No sentient code please
... kidding, totally optional suggestion